### ISSUE TITLE
Unable to upgrade on Ubuntu 16.04

### ISSUE THREAD

**Issue by watou** [digitalocean/debian-sys-maint-roll-passwd#20]
I performed this hotfix last month, but now attempting to apt-get upgrade MySQL server, the upgrade hangs, eventually producing this message:

```
mysql_upgrade: Got error: 1045: Access denied for user 'debian-sys-maint'@'localhost' (using password: YES) while connecting to the MySQL server
Upgrade process encountered error and will not continue.
mysql_upgrade failed with exit status 11
```

Also I get access denied when trying to connect to MySQL using the credentials in the patched /etc/debian/mysql.cnf:

```
# Automatically generated for Debian scripts. DO NOT TOUCH!
# This was updated via a 1-Click HotFix on Mon, 18 Sep 2017 17:49:45 +0100
[client]
host     = localhost
user     = debian-sys-maint
password = redacted
socket   = /var/run/mysqld/mysqld.sock
[mysql_upgrade]
host     = localhost
user     = debian-sys-maint
password = redacted
socket   = /var/run/mysqld/mysqld.sock
```

using the command:

```
mysql -u debian-sys-maint -p [and entering the password that's in /etc/mysql/debian.cnf]
```

What to do to recover from this?  Was this situation created by the hotfix?

I notice on [this line](https://github.com/digitalocean/debian-sys-maint-roll-passwd/blob/master/fix.sh#L88) that the query it attempting to change a column called `password` in the `user` table of the `mysql` database, but mine only has a column named `authentication_string`.  So the hotfix might have failed to update the database with the password that was successfully written to the file `/etc/mysql/debian.cnf`.  WDYT?

**Comment by watou** [comment]
In my particular installation, I was able to get past the `debian-sys-maint` password mismatch this way:
1. copy the password in the hotfixed `/etc/mysql/debian.cnf` to the clipboard.
1. `mysql -u root -p` using the root password for MySQL (on my droplet, it's the same as the root password for the droplet).
1. `use mysql;`
1. `SET GLOBAL validate_password_mixed_case_count = 0;`
1. `SET GLOBAL validate_password_special_char_count = 0;`
1. `ALTER USER 'debian-sys-maint'@'localhost' IDENTIFIED BY '<paste password from clipboard>';`
1. `quit`
1. `sudo apt-get upgrade` now worked.

**Comment by jeremylind** [comment]
This worked for me, thank you so much! Is it possible that the hotfix script doesn't take into account unique password requirements and fails to update the `debian-sys-maint` password in mysql, but still updates it in `/etc/mysql/debian.cnf`?

**Comment by watou** [comment]
>  Is it possible that the hotfix script doesn't take into account unique password requirements and fails to update the debian-sys-maint password in mysql, but still updates it in /etc/mysql/debian.cnf?

Yes, and also that the script attempts to update the `user` table directly, while only `ALTER USER` should be used (hiding if the column is called `password` or `authentication_string`).
